pkgs/tools/networking/isync/default.nix

```diff
@@ -20,10 +21,16 @@ stdenv.mkDerivation rec {
     ./work-around-unexpected-EOF-error-messages-at-end-of-SSL-connections.patch
   ];
 
-  nativeBuildInputs = [ pkg-config perl ];
+  nativeBuildInputs = [ pkg-config perl ]
+    ++ lib.optionals withCyrusSaslXoauth2 [ makeWrapper ];
   buildInputs = [ openssl db cyrus_sasl zlib ]
     ++ lib.optionals stdenv.isDarwin [ Security ];
 
+  postInstall = lib.optionalString withCyrusSaslXoauth2 ''
+    wrapProgram "$out/bin/mbsync" \
+        --prefix SASL_PATH : "${cyrus_sasl}/lib/sasl2:${cyrus-sasl-xoauth2}/lib/sasl2"
```

Comment:
Can we use lib.makeLibPath or similar here?

Reply:
Sure. And by doing that, I figured out that the `${cyrus_sasl}/lib/sasl2` part is not needed at all, becasue `cyrus_sasl` it has no `lib/sasl2` (originally, I copied it from some issue discussion). However, I kept the call to `makeSearchPath` for the case more libraries will be needed in the future.

The PR is updated.